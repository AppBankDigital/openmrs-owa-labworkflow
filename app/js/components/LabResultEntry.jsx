/* * This Source Code Form is subject to the terms of the Mozilla Public License,
 * v. 2.0. If a copy of the MPL was not distributed with this file, You can
 * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
 * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
 *
 * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
 * graphic logo is a trademark of OpenMRS Inc.
 */
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Loader } from '@openmrs/react-components';
import { formValueSelector } from 'redux-form';
import R from 'ramda';

import {
  Grid,
  Row,
  FormGroup,
} from 'react-bootstrap';
import moment from 'moment';
import { Redirect } from 'react-router-dom';

import {
  CustomDatePicker,
  PatientHeader,
  EncounterFormPage,
  Obs,
  formValidations,
} from '@openmrs/react-components';

import patientAction from '../actions/patientAction';
import { fetchLabConcept } from '../actions/labConceptsAction';
import '../../css/lab-result-entry.scss';
import { formatRangeDisplayText, hasMaxAndMinValues } from '../utils/helpers';


const {
  minValue,
  maxValue,
  abnormalMaxValue,
  abnormalMinValue,
  maxDateValue,
} = formValidations;

export class LabResultEntry extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      patientHeaderDetail: false,
      redirect: false,
    };
  }

  componentWillMount() {
    const { dispatch, history: { location: { state } } } = this.props;
    if (typeof state !== 'undefined') {
      const conceptUUID = state.concept.uuid;
      dispatch(patientAction.getPatient(state.patient.uuid));
      dispatch(fetchLabConcept(conceptUUID));
    } else {
      this.shouldRedirect();
    }
  }

  componentWillReceiveProps(nextProps) {
    const { patientHeaderDetail } = this.props;

    if (nextProps.patientHeaderDetail.uuid !== patientHeaderDetail.uuid) {
      this.setState({
        patientHeaderDetail: nextProps.patientHeaderDetail,
      });
    }
  }

  shouldRedirect() {
    this.setState({ redirect: true });
  }

  renderForm(selectedLabConcept) {
    const { patientHeaderDetail } = this.state;
    const {
      CONSTANTS, conceptMembers, didNotPerformCheckbox, history: { location: { state } },
    } = this.props;

    const encounterType = {
      uuid: CONSTANTS.labResultsEncounterType,
    };

    const hasMembers = selectedLabConcept.set;
    const hasAnswers = !!selectedLabConcept.answers.length;
    const isSingle = !hasMembers && !selectedLabConcept.set && !hasAnswers;
    let hasRanges = false;

    if (hasMembers && !R.isEmpty(conceptMembers)) {
      hasRanges = hasMaxAndMinValues(selectedLabConcept.setMembers, conceptMembers);
    } else if (!hasAnswers) {
      hasRanges = selectedLabConcept.hiNormal !== 'null' && selectedLabConcept.lowNormal !== 'null';
    }

    const maxDateRange = maxDateValue(new Date());
    const observations = (
      <Grid>
        {
          (hasMembers || hasAnswers || isSingle)
            && (
              <div className="col-xs-4">
                <span className="single-result-field">
                  <span className="obs-date-label">Result Date</span>
                  <span className="obs-date-field">
                    <Obs
                      datatype="date"
                      concept="68d6bd27-37ff-4d7a-87a0-f5e0f9c8dcc0"
                      path="result-date"
                      validate={[maxDateRange]}
                    />
                  </span>
                </span>
              </div>
            )
        }
        <div className="col-xs-4">
          <div className="did-not-perform-checkbox">
            <Obs
              conceptAnswer={CONSTANTS.labResultsDidNotPerformAnswer}
              widget="checkbox"
              concept={CONSTANTS.labResultsDidNotPerformQuestion}
              path="did-not-perform-checkbox"
              checkBoxTitle="Did not perform"
            />
          </div>
        </div>
        {didNotPerformCheckbox && (
          <div className="col-xs-4">
            <div className="did-not-perform">
              <span className="did-not-perform-label">Reason:&nbsp;</span>
              <Obs
                conceptAnswers={["Low Volume",
                  "Contaminated",
                  "Equipment Failure",
                  "Stock Out",
                  "Blood Coagulation",
                  "Inappropriate Container"]}
                widget="dropdown"
                concept={CONSTANTS.fetchLabResultsDidNotPerformReason}
                path="did-not-perform-dropdown"
              />
            </div>
          </div>
        )}
        {
          (hasAnswers)
            && (
              <div className="col-xs-4">
                <span className="single-result-field">
                  <span className="obs-dropdown-label">{selectedLabConcept.display}</span>
                  <span className="obs-dropdown-field">
                    <Obs
                      conceptAnswers={selectedLabConcept.answers}
                      widget="dropdown"
                      concept={selectedLabConcept.uuid}
                      path={selectedLabConcept.uuid}
                    />
                  </span>
                </span>
              </div>
            )
        }
        <Row>
          {(hasMembers)
            && (
              selectedLabConcept.setMembers.map(
                member => this.renderFormContent(member),
              ))
          }
          {(isSingle)
          && (this.renderFormContent(selectedLabConcept))
          }
        </Row>
      </Grid>
    );

    return (
      <div className="fieldset-container lab-entry-result-details">
        <div className="legend">
          <span> Result Details </span>
        </div>
        <div className="fieldset-body">
          {hasRanges
            && (<span className="range-header-text">NORMAL RANGE</span>)
          }
          <span className="encounter-form-componnent">
            <EncounterFormPage
              defaultValues={[
                {
                  type: "obs",
                  path: "test-order-number",
                  concept: "393dec41-2fb5-428f-acfa-36ea85da6666",
                  value: state.orderNumber,
                },
              ]}
              afterSubmitLink="/"
              backLink="/"
              encounterType={encounterType}
              formContent={observations}
              patient={patientHeaderDetail}
              formId="result-entry-form"
            />
          </span>
        </div>
      </div>
    );
  }

  renderFormContent(member) {
    const { conceptMembers } = this.props;
    const memberDetails = conceptMembers[member.uuid] || member;
    const {
      hiNormal,
      lowNormal,
      lowAbsolute,
      hiAbsolute,
      hiCritical,
      lowCritical,
      units,
    } = memberDetails;

    let obsProps = {
      concept: member.uuid,
      path: member.uuid,
    };

    const validations = [];
    const warnings = [];

    if (lowAbsolute) {
      const minRange = minValue(lowAbsolute);
      validations.push(minRange);
    }

    if (hiAbsolute) {
      const maxRange = maxValue(hiAbsolute);
      validations.push(maxRange);
    }

    if (hiCritical) {
      const abnormalMaxRange = abnormalMaxValue(hiCritical);
      warnings.push(abnormalMaxRange);
    }

    if (lowCritical) {
      const abnormalMinRange = abnormalMinValue(lowCritical);
      warnings.push(abnormalMinRange);
    }

    if (!R.isEmpty(validations)) {
      obsProps = {
        ...obsProps,
        validate: validations,
      };
    }

    if (!R.isEmpty(warnings)) {
      obsProps = {
        ...obsProps,
        warn: warnings,
      };
    }

    const normalRange = formatRangeDisplayText(lowNormal, hiNormal, units);
    return (
      <FormGroup controlId={member.display}>
        <span className="member-display-label">
          {member.display}
        </span>
        <span className="obs-component">
          <Obs
            {...obsProps}
          />
        </span>
        <span className="units">{units || ''}</span>
        <span className="valid-range">
          {normalRange}
        </span>
      </FormGroup>
    );
  }

  render() {
    const { patientHeaderDetail, redirect } = this.state;
    const {
      location, selectedLabConcept,
    } = this.props;

    if (!location.state || redirect) {
      return <Redirect to="/" />;
    }
    if (!R.isEmpty(selectedLabConcept)) {
      return (
        <div className="container-fluid">
          {patientHeaderDetail && <PatientHeader patient={patientHeaderDetail} />}
          {location.state
          && (
            <div>
              <h2 className="lab-entry-page-title">Test Results</h2>
              <div className="fieldset-container">
                <div className="legend">
                  <span> Specimen Details </span>
                </div>
                <div className="fieldset-body">
                  <div className="col-xs-12">
                    <CustomDatePicker
                      handleDateChange={() => {}}
                      labelClassName="date-picker-label"
                      label="Specimen Collection Date:"
                      defaultDate={moment().subtract(7, 'days')}
                      field="dateFromField"
                    />
                  </div>
                  <br />
                  <br />
                  <div className="col-xs-6">
                    <span className="test-details-label">
                  Test:&nbsp;
                      <span className="test-details">{location.state.display}</span>
                    </span>
                  </div>
                  <div className="col-xs-6">
                    <span className="test-details-label">
                  Urgency:&nbsp;
                      <span className="test-details">{location.state.urgency}</span>
                    </span>
                  </div>
                  <br />
                  <br />
                </div>
              </div>
            </div>
          )}
          <br />
          <br />
          <div>
            {this.renderForm(selectedLabConcept)}
          </div>
        </div>
      );
    }

    return (
      <Loader />
    );
  }
}

LabResultEntry.defaultProps = {
  selectedLabConcept: null,
};

LabResultEntry.propTypes = {
  patientHeaderDetail: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  selectedLabConcept: PropTypes.object,
  location: PropTypes.object.isRequired,
  CONSTANTS: PropTypes.object.isRequired,
  conceptMembers: PropTypes.object.isRequired,
  didNotPerformCheckbox: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => {
  const {
    patient: { patient },
    selectedLabConcept,
    openmrs: { CONSTANTS },
    conceptMembers,
  } = state;
  const selector = formValueSelector('result-entry-form');
  const didNotPerformCheckbox = CONSTANTS && !!(selector(state, `obs|path=did-not-perform-checkbox|concept=${CONSTANTS.labResultsDidNotPerformQuestion}`));

  return {
    patientHeaderDetail: patient,
    selectedLabConcept,
    CONSTANTS,
    conceptMembers,
    didNotPerformCheckbox,
  };
};

export default connect(mapStateToProps)(LabResultEntry);
