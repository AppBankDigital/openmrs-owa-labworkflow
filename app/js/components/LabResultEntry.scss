legend {
    width: auto;
    font-size: 15px;
}

fieldset {
    border: 1px solid black;
}

.result-input {
    width: 70px;
    border: 1px solid black;
    padding: 0px 5px;
}

.result-detail-range {
    font-size: 13px;
}

.fieldset-container {
    background-color: lightgrey;
}

input[type="checkbox"] {
    transform: scale(2);
}

span.test-details {
  font-size: 20px;
}

a.link-cancel {
  color: white
}

.swal-button--NO {
  background: #f66;
 }
.swal-button--YES,
.swal-button--confirm {
  background: #8eb82a;
} 

button.pull-right.btn-success {
  background: #5cb85c;
}

.date-picker-label {
  line-height: 35px;
  margin-right: 10px;
  font-size: 14px;
  width: 100%;
}
