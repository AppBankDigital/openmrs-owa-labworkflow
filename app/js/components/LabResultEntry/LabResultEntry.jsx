import React, { PureComponent } from 'react';


class LabResultEntry extends PureComponent {
  render() {
    const { location } = this.props;
    const { patient } = location.state;
    const displayText = `Lab Results for ${patient.display}`;
    return (
      <div>
        <h1>{displayText}</h1>
      </div>
    )
  }
}

export default LabResultEntry;