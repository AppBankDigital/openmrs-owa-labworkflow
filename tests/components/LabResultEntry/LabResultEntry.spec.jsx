import React from 'react';
import { HashRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import LabResultEntry from '../../../app/js/components/LabResultEntry';


let mountedComponent;


let props = {
  patientHeaderDetail: {
    display: "Y2A7LR - williams willy john",
  },
  conceptMembers: {
    '1234-abcd': {
      hiNormal: 22,
      lowNormal: 11,
      lowAbsolute: 9,
      hiAbsolute: 25,
      hiCritical: 30,
      lowCritical: 3,
      units: 'cc',
    },
    'pqrs-5678': {
      hiNormal: 22,
      lowNormal: 11,
      lowAbsolute: 9,
      hiAbsolute: 25,
      hiCritical: 30,
      lowCritical: 3,
      units: 'cc',
    },
  },
  clearNameEMRField: jest.fn(),
  labTests: ['Hémogramme automatisé'],
  nameField: '',
  location: {
    state: {
      orderNumber: "ORD-1",
      concept: {
        display: "Hémogramme automatisé",
      },
      urgency: 'ROUTINE',
      dateActivated: "2018-08-30T17:34:19.000+0100",
      patient: {
        display: "Y2A7LR - williams willy john",
      },
      auditInfo: {
        dateCreated: "2018-08-30T17:34:19.000+0100",
      },
    },
  },
  history: {
    location: {
      state: {
        orderNumber: "ORD-1",
        concept: {
          uuid: '9i9i-99090-999',
          display: "Hémogramme automatisé",
        },
        urgency: 'ROUTINE',
        dateActivated: "2018-08-30T17:34:19.000+0100",
        patient: {
          display: "Y2A7LR - williams willy john",
        },
      },
    },
  },
};

const state = {
  patient: {
    patient: {
      display: "Y2A7LR - williams willy john",
    },
  },
  openmrs: {
    patients: {
      abcd: {
        uuid: 'abcd',
        display: "Y2A7LR - williams willy john",
      },
    },
    selectedPatient: 'abcd',
    CONSTANTS: {
      labResultsTestOrderNumberConcept: 'mock-test-orderno-concept',
      labResultsDateConcept: 'mock-date-concept',
      labResultsEncounterType: 'mock-encounter-uuid',
      labResultsDidNotPerformAnswer: 'mock-uuid',
      labResultsDidNotPerformQuestion: 'mock-uuid',
      labResultsDidNotPerformReason: 'mock-uuid',
    },
  },
  form: {},
  selectedLabConcept: {
    answers: [],
    set: true,
    setMembers: [
      {
        uuid: '1234-abcd',
        display: "Height",
      },
      {
        uuid: 'pqrs-5678',
        display: "Weight",
      },
    ],
  },
  conceptMembers: {
    '1234-abcd': {
      hiNormal: 22,
      lowNormal: 11,
      lowAbsolute: 9,
      hiAbsolute: 25,
      hiCritical: 30,
      lowCritical: 3,
      units: 'cc',
    },
    'pqrs-5678': {
      hiNormal: 22,
      lowNormal: 11,
      lowAbsolute: 9,
      hiAbsolute: 25,
      hiCritical: 30,
      lowCritical: 3,
      units: 'cc',
    },
  },
};

const store = mockStore(state);
const context = {
  store,
  storeSubscription: {
    onStateChange: jest.fn,
    parentSub: null,
    store,
    unsubscribe: jest.fn,
  },
};

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <HashRouter>
          <LabResultEntry {...props} />
        </HashRouter>
      </Provider>,
      { context },
    );
  }
  return mountedComponent;
};


describe('<LabResultEntry /> component', () => {
  it('renders correctly with valid props', (done) => {
    const component = getComponent();
    component.setProps({ location: { state: {} } });
    expect(component).toMatchSnapshot();
    done();
  });

  it('should render <Loader /> component when selectedLabConcept is empty', () => {
    props = {
      ...props,
      selectedLabConcept: {},
    };
    const store = mockStore({
      patient: {
        patient: {
          display: "Y2A7LR - williams willy john",
        },
      },
      selectedLabConcept: {},
      openmrs: {
        CONSTANTS: {
          labResultsTestOrderNumberConcept: 'mock-test-orderno-concept',
          labResultsDateConcept: 'mock-date-concept',
          labResultsEncounterType: 'mock-encounter-uuid',
          labResultsDidNotPerformAnswer: 'mock-uuid',
          labResultsDidNotPerformQuestion: 'mock-uuid',
          labResultsDidNotPerformReason: 'mock-uuid',
        },
      },
    });
    const component = mount(<LabResultEntry {...props} store={store} />);
    expect(component).toMatchSnapshot();
  });

  it('should redirect when location has an empty state', () => {
    props = {
      ...props,
      location: {
        state: null,
      },
    };
    const store = mockStore({
      patient: {
        patient: {
          display: "Y2A7LR - williams willy john",
        },
      },
      openmrs: {
        CONSTANTS: {
          labResultsTestOrderNumberConcept: 'mock-test-orderno-concept',
          labResultsDateConcept: 'mock-date-concept',
          labResultsEncounterType: 'mock-encounter-uuid',
          labResultsDidNotPerformAnswer: 'mock-uuid',
          labResultsDidNotPerformQuestion: 'mock-uuid',
          labResultsDidNotPerformReason: 'mock-uuid',
        },
      },
    });
    const component = mount(
      <HashRouter>
        <LabResultEntry {...props} store={store} />
      </HashRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
