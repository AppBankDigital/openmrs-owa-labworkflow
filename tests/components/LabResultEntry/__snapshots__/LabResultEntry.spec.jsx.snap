// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabOrdersList /> component renders correctly 1`] = `
<Connect(LabResultEntry)
  clearNameEMRField={[MockFunction]}
  history={
    Object {
      "location": Object {
        "state": Object {
          "concept": Object {
            "display": "Hémogramme automatisé",
          },
          "dateActivated": "2018-08-30T17:34:19.000+0100",
          "orderNumber": "ORD-1",
          "patient": Object {
            "display": "Y2A7LR - williams willy john",
          },
          "urgency": "ROUTINE",
        },
      },
    }
  }
  labTests={
    Array [
      "Hémogramme automatisé",
    ]
  }
  location={
    Object {
      "state": Object {
        "concept": Object {
          "display": "Hémogramme automatisé",
        },
        "dateActivated": "2018-08-30T17:34:19.000+0100",
        "orderNumber": "ORD-1",
        "patient": Object {
          "display": "Y2A7LR - williams willy john",
        },
        "urgency": "ROUTINE",
      },
    }
  }
  nameField=""
  patientHeaderDetail={
    Object {
      "display": "Y2A7LR - williams willy john",
    }
  }
  selectedLAbConcept={
    Object {
      "setMembers": Array [
        Object {
          "display": "Height",
          "uuid": "1234-abcd",
        },
        Object {
          "display": "Weight",
          "uuid": "pqrs-5678",
        },
      ],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LabResultEntry
    clearNameEMRField={[MockFunction]}
    dispatch={[Function]}
    history={
      Object {
        "location": Object {
          "state": Object {
            "concept": Object {
              "display": "Hémogramme automatisé",
            },
            "dateActivated": "2018-08-30T17:34:19.000+0100",
            "orderNumber": "ORD-1",
            "patient": Object {
              "display": "Y2A7LR - williams willy john",
            },
            "urgency": "ROUTINE",
          },
        },
      }
    }
    labTests={
      Array [
        "Hémogramme automatisé",
      ]
    }
    location={
      Object {
        "state": Object {
          "concept": Object {
            "display": "Hémogramme automatisé",
          },
          "dateActivated": "2018-08-30T17:34:19.000+0100",
          "orderNumber": "ORD-1",
          "patient": Object {
            "display": "Y2A7LR - williams willy john",
          },
          "urgency": "ROUTINE",
        },
      }
    }
    nameField=""
    patientHeaderDetail={
      Object {
        "display": "Y2A7LR - williams willy john",
      }
    }
    selectedLAbConcept={
      Object {
        "setMembers": Array [
          Object {
            "display": "Height",
            "uuid": "1234-abcd",
          },
          Object {
            "display": "Weight",
            "uuid": "pqrs-5678",
          },
        ],
      }
    }
    selectedLabConcept={null}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="container-fluid"
    >
      <div>
        <h1>
          Test Results
        </h1>
        <div
          className="fieldset-container"
        >
          <fieldset>
            <legend>
              Specimen Details:
            </legend>
            <div
              className="col-xs-12"
            >
              <CustomDatePicker
                defaultDate={"2018-09-12T18:19:27.012Z"}
                field="dateFromField"
                formControlStyle={
                  Object {
                    "marginRight": "5px",
                    "width": "100px",
                  }
                }
                handleDateChange={[Function]}
                label="Specimen Collection Date:"
                labelClassName="date-picker-label"
              >
                <DatePicker
                  allowSameDay={false}
                  customInput={<DateDisplayComponent />}
                  dateFormat="L"
                  dateFormatCalendar="MMMM YYYY"
                  disabled={false}
                  disabledKeyboardNavigation={false}
                  dropdownMode="scroll"
                  monthsShown={1}
                  nextMonthButtonLabel="Next month"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClickOutside={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMonthChange={[Function]}
                  onSelect={[Function]}
                  onYearChange={[Function]}
                  preventOpenOnFocus={false}
                  previousMonthButtonLabel="Previous Month"
                  readOnly={false}
                  selected={"2018-09-12T18:19:27.012Z"}
                  shouldCloseOnSelect={true}
                  showTimeSelect={false}
                  timeCaption="Time"
                  timeIntervals={30}
                  withPortal={false}
                >
                  <PopperComponent
                    hidePopper={true}
                    popperComponent={null}
                    popperModifiers={
                      Object {
                        "preventOverflow": Object {
                          "boundariesElement": "viewport",
                          "enabled": true,
                          "escapeWithReference": true,
                        },
                      }
                    }
                    popperPlacement="bottom-start"
                    targetComponent={
                      <div
                        className="react-datepicker__input-container"
                      >
                        <DateDisplayComponent
                          className=""
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          readOnly={false}
                          value="09/12/2018"
                        />
                      </div>
                    }
                  >
                    <Manager
                      tag="div"
                    >
                      <div>
                        <Target
                          className="react-datepicker-wrapper"
                        >
                          <div
                            className="react-datepicker-wrapper"
                          >
                            <div
                              className="react-datepicker__input-container"
                            >
                              <DateDisplayComponent
                                className=""
                                disabled={false}
                                onBlur={[Function]}
                                onChange={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                readOnly={false}
                                value="09/12/2018"
                              >
                                <span
                                  style={
                                    Object {
                                      "display": "flex",
                                      "justifyContent": "space-around",
                                    }
                                  }
                                >
                                  <span
                                    className="date-picker-label"
                                  >
                                    Specimen Collection Date:
                                  </span>
                                  <FormControl
                                    bsClass="form-control"
                                    componentClass="input"
                                    placeholder=""
                                    style={
                                      Object {
                                        "marginRight": "5px",
                                        "width": "100px",
                                      }
                                    }
                                    type="text"
                                    value="09/12/2018"
                                  >
                                    <input
                                      className="form-control"
                                      placeholder=""
                                      style={
                                        Object {
                                          "marginRight": "5px",
                                          "width": "100px",
                                        }
                                      }
                                      type="text"
                                      value="09/12/2018"
                                    />
                                  </FormControl>
                                  <i
                                    className="icon-calendar scale small add-on"
                                    onClick={[Function]}
                                    role="toolbar"
                                  />
                                </span>
                              </DateDisplayComponent>
                            </div>
                          </div>
                        </Target>
                      </div>
                    </Manager>
                  </PopperComponent>
                </DatePicker>
              </CustomDatePicker>
            </div>
            <br />
            <br />
            <div
              className="col-xs-6"
            >
              <span>
                Test: 
                <span
                  className="test-details"
                />
              </span>
            </div>
            <div
              className="col-xs-6"
            >
              <span>
                Urgency: 
                <span
                  className="test-details"
                >
                  ROUTINE
                </span>
              </span>
            </div>
            <br />
            <br />
          </fieldset>
        </div>
      </div>
      <br />
      <br />
      <br />
      <div
        className="fieldset-container"
      />
    </div>
  </LabResultEntry>
</Connect(LabResultEntry)>
`;
